expression/builtin_cast.go

```diff
@@ -464,7 +464,7 @@ func (b *builtinCastIntAsRealSig) evalReal(row chunk.Row) (res float64, isNull b
 		res = 0
 	} else {
 		var uVal uint64
-		uVal, err = types.ConvertIntToUint(val, types.UnsignedUpperBound[mysql.TypeLonglong], mysql.TypeLonglong)
+		uVal, err = types.ConvertIntToUint(nil, val, types.UnsignedUpperBound[mysql.TypeLonglong], mysql.TypeLonglong)
```

Comment:
Why not pass `sc` here?

Reply:
@XuHuaiyu To my understanding, this function is used for functions like `CAST`, and there is no difference whether we set `sql_mode=''`:
```sql
mysql> select cast(-1 as unsigned);
+----------------------+
| cast(-1 as unsigned) |
+----------------------+
| 18446744073709551615 |
+----------------------+
1 row in set (0.55 sec)

mysql> set sql_mode='';
Query OK, 0 rows affected, 1 warning (0.00 sec)

mysql> select cast(-1 as unsigned);
+----------------------+
| cast(-1 as unsigned) |
+----------------------+
| 18446744073709551615 |
+----------------------+
1 row in set (0.00 sec)
```
So no need to pass `sc` to check.

Reply:
Reasonable. I was using this `nil` to differ `CAST` from this case. It turns out that there is no need to do that. Nice catch :-)